content/en/docs/Advanced/sys-extensions.md

```diff
@@ -0,0 +1,156 @@
+
+We recommend bundling the sysexts during the build process, but this is an alternative for those cases where the sysexts are not known at build time or need to be added later to an existing system.
+
+This is also a good choice for testing sysexts before bundling them into the install media as they can be added to just one entry in the system and tested without affecting other entries.
+
+
+### Boot workflow
+
+During boot, systemd-stub will copy and measure the system extensions from the proper directory according to the loaded EFI file (`/EFI/kairos/$EFI_FILE.efi.extra.d/`) and copy them into the initrd `/.extra/sysext` directory.
+Immucore then will extract the public keys from the firmware and save them under `/run/verity.d` so the systemd extensions can be veried against those.
+Immucore will then verify those sysexts to see if they are signed and verity protected. If they are, they will be copied under `/run/extensions` and during the `initramfs` yip stage the `systemd-sysext` service will be run and they will be mounted in their proper directories.
+
+The enablement of the system extensions service is done at the last step in the `initramfs` stage to not collide with anything in those stage writing to the directories under `/usr/local` (mounted from persistent partition), as some dirs get mounted as read only, they could collide with the stages writing to those dirs.
+
+So if using binaries from the system extensions is needed during boot, make sure they are done after the `initramfs` stage. Otherwise they wont be available in earlier stages.
+
+
+### Known issues
+
+- Sysext images need to be named with the extension `.sysext.raw` to be identified correctly. This is a design choice to avoid conflicts with other files that could be present in the EFI partition and we dont expect this to change int he future.
+- Any folder that is mounted as a system extension will be mounted as read-only. So if your sysext is mounting `/usr/local/bin` to add binaries, it will be mounted as read-only. Other sysexts can eb added and they will be merged correctly, but the final dir will be read-only. This is a limitation of the current systemd version (lower than 256) and will be addressed in future releases.
+- Only `/usr` can be extended. This is a design choice and might change in the future to allow other directories to be extended.
+- System extensions provided binaries are only available after the `initramfs` stage.
+- Any extensions bundled on the install media will be available in the `active` and `passive` boot entries. Any other custom entries will need to have the sysexts copied manually to the proper directory in the EFI partition.
+- `recovery` and `autoreset` boot entries have no extensions nor are planned to have. This is a design choice to keep the recovery environment as minimal as possible without anything that could break it.
```

Comment:
`This is a design choice` : ours or systemd? Same thing for item 1 above.

Reply:
should I clarify this further?

Reply:
```suggestion
- `recovery` and `autoreset` boot entries have no extensions nor are planned to have. This is a Kairos design choice to keep the recovery environment as minimal as possible without anything that could break it.
```